lambdas/es/indexer/index.py

```diff
@@ -1,74 +1,218 @@
 """
-phone data into elastic for supported file extensions
+phone data into elastic for supported file extensions.
+note: we truncated inbound documents to no more than DOC_SIZE_LIMIT characters
+(this bounds memory pressure and request size to elastic)
 """
+
 from datetime import datetime
+from math import floor
 import json
 import os
-from urllib.parse import unquote
+from urllib.parse import unquote, unquote_plus
 
 from aws_requests_auth.aws_auth import AWSRequestsAuth
-import botocore
 import boto3
+import botocore
 from elasticsearch import Elasticsearch, RequestsHttpConnection
-from elasticsearch.exceptions import RequestError
+from elasticsearch.helpers import bulk
 import nbformat
-import tenacity
-
-DEFAULT_CONFIG = {
-    'to_index': [
-        '.ipynb',
-        '.md',
-        '.rmd',
-    ]
-}
+from tenacity import stop_after_attempt, stop_after_delay, retry, wait_exponential
 
+CONTENT_INDEX_EXTS = [
+    ".csv",
+    ".html",
+    ".ipynb",
+    ".json",
+    ".md",
+    ".rmd",
+    ".txt",
+    ".xml"
+]
+# 10 MB, see https://amzn.to/2xJpngN
+CHUNK_LIMIT_BYTES = 20_000_000
+DOC_LIMIT_BYTES = 10_000
+ELASTIC_TIMEOUT = 30
+MAX_RETRY = 10 # prevent long-running lambdas due to malformed calls
 NB_VERSION = 4 # default notebook version for nbformat
+# signifies that the object is truly deleted, not to be confused with
+# s3:ObjectRemoved:DeleteMarkerCreated, which we may see in versioned buckets
+# see https://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html
+OBJECT_DELETE = "ObjectRemoved:Delete"
+QUEUE_LIMIT_BYTES = 300_000_000# 100MB
+RETRY_429 = 5
+TEST_EVENT = "s3:TestEvent"
+# we need to filter out GetObject and HeadObject calls generated by the present
+#  lambda in order to display accurate analytics in the Quilt catalog
+#  a custom user agent enables said filtration
+USER_AGENT = "quilt3-python"
 
-S3_CLIENT = boto3.client("s3")
+def bulk_send(elastic, list_):
+    """make a bulk() call to elastic"""
+    return bulk(
+        elastic,
+        iter(list_),
```

Comment:
Can you see if it works without `iter`?

The docs say "iterator"... however, that function then calls `streaming_bulk`, for which the docs say "iterable". One of them is clearly wrong.